### PR TITLE
Display better information when running 'oc'

### PR DESCRIPTION
**Pull request by smarterclayton**
Add a "welcome" type page to 'oc', and make 'oc help' shorter. Clean up
where certain commands are registered in 'oc help'. Add a few more
descriptions to 'oc types'.

[test]

```
$ oc
OpenShift Client

This client helps you develop, build, deploy, and run your applications on any OpenShift or
Kubernetes compatible platform. It also includes the administrative commands for managing a
cluster under the 'adm' subcommand.

To create a new application, login to your server and then run new-app:

  $ oc login https://myorigincluster.com
  $ oc new-app centos/ruby-22-centos7~https://github.com/openshift/ruby-hello-world.git

This will create an application based on the Docker image 'centos/ruby-22-centos7' that builds
the source code from GitHub. A build will start automatically, push the resulting image to the
registry, and a deployment will roll that change out in your application.

Once your application is deployed, use the status, describe, and log commands to see more about
the created components:

  $ oc status
  $ oc describe deploymentconfig ruby-hello-world
  $ oc logs dc/ruby-hello-world

To make this application visible outside of the cluster, use the expose command on the service
we just created to create a 'route' (which will connect your application over the HTTP port
to a public domain name).

  $ oc expose svc/ruby-hello-world
  $ oc status

You should now see the URL the application can be reached at.

To see the full list of commands supported, run 'oc help'.
```

### CODE REVIEW

**Comment by smarterclayton** [comment]
@fabianofranz

File: pkg/cmd/cli/cli.go [hunk 1]
```diff
@@ -26,41 +26,56 @@ import (
...
+
+  $ %[1]s login https://mycluster.mycompany.com
   $ %[1]s new-app centos/ruby-22-centos7~https://github.com/openshift/ruby-hello-world.git
+  $ %[1]s logs -f bc/ruby-hello-world
 
 This will create an application based on the Docker image 'centos/ruby-22-centos7' that builds
-the source code at 'github.com/openshift/ruby-hello-world.git'. A build will start automatically and
-a deployment will start as soon as the build finishes.
+the source code from GitHub. A build will start automatically, push the resulting image to the
+registry, and a deployment will roll that change out in your application.
```
Comment (fabianofranz) [inline pkg/cmd/cli/cli.go]: "...roll that change out in your project." maybe?

File: pkg/cmd/cli/cli.go [hunk 2]
```diff
@@ -26,41 +26,56 @@ import (
...
 
-Once your application is deployed, use the status, get, and describe commands to see more about
+Once your application is deployed, use the status, describe, and log commands to see more about
```
Comment (fabianofranz) [inline pkg/cmd/cli/cli.go]: The line mentions `log` but there isn't an example below.

**Comment by fabianofranz** [comment]
Minor nits, LGTM

**Comment by smarterclayton** [comment]
Updated, [merge]

**Pull request merged**